cloudinit/net/dhcp.py

```diff
@@ -167,7 +168,7 @@ def __init__(self):
             raise NoDHCPLeaseMissingDhclientError()
 
     @staticmethod
-    def parse_dhcp_lease_file(lease_file):
+    def parse_dhcp_lease_file(lease_file: str) -> list:
```

Comment:
I think we may want to `from typing import Any, Dict, List` and have to following:
```suggestion
    def parse_dhcp_lease_file(lease_file: str) -> List[Dict[str, Any]]:
```